lib/mj-single.js

```diff
@@ -570,7 +570,22 @@ function GetHTML(result) {
     for (var i=0, m=html.childNodes.length; i < m; i++)
       html.childNodes[i].setAttribute("aria-hidden",true);
   }
-
+  // remove automatically generated IDs
+  var ids = html.querySelectorAll('[id^="MJXc-Node-"]');
+  for (var i = 0; i < ids.length; i++){
+    ids[i].removeAttribute("id");
+  }
+  // remove extreneous frame element
+  var frame = html.querySelector('[id^="MathJax-Element-"]');
+  if (frame){
+    // in display-mode, the frame is inside the display-style wrapper
+    html.insertBefore(frame.firstChild, frame);
+    html.removeChild(frame);
+  }
+  else{
+    // otherwise (inline-mode) the frame is the root element
+    html.removeAttribute("id");
```

Comment:
They probably can, but I haven't done any testing to check for sure.  Personally, I'd leave the outer display element rather than move the class to the inner one, but it would probably still work if you did.